diff --git a/pkg/tcpip/stack/save_restore.go b/pkg/tcpip/stack/save_restore.go
index 48ed8c782..96e131612 100644
--- a/pkg/tcpip/stack/save_restore.go
+++ b/pkg/tcpip/stack/save_restore.go
@@ -16,8 +16,8 @@ package stack
 
 import (
 	"context"
+	"encoding/binary"
 	"math/rand"
-	"time"
 
 	cryptorand "gvisor.dev/gvisor/pkg/rand"
 )
@@ -49,7 +49,10 @@ func (s *Stack) beforeSave() {
 
 // afterLoad is invoked by stateify.
 func (s *Stack) afterLoad(context.Context) {
-	s.insecureRNG = rand.New(rand.NewSource(time.Now().UnixNano()))
+	// Use DST-seeded PRNG for deterministic snapshot restore.
+	var seed int64
+	binary.Read(cryptorand.Reader, binary.LittleEndian, &seed)
+	s.insecureRNG = rand.New(rand.NewSource(seed))
 	s.secureRNG = cryptorand.RNGFrom(cryptorand.Reader)
 }
